targets/actionscript/make.js

```diff
@@ -184,7 +239,13 @@ function getModelPropertyInit(tabbing, property, datatype) {
             else
                 throw "Unknown collection type: " + property.collection + " for " + property.name + " in " + datatype.name;
         } else {
-            return tabbing + property.name + " = new " + property.actualtype + "(data." + property.name + ");";
+            if (fullNamespaceRequired(property.actualtype)) {
+                return tabbing + property.name + " = new " + getValidPlayFabActionScriptNamespacePrefix(datatype) + property.actualtype + "(data." + property.name + ");";
```

Comment:
I'm almost thinking that
getValidPlayFabActionScriptNamespacePrefix
can just return empty string when !fullNamespaceRequired
That way you can simplify this block change.